source/python/idetect/nlp_models/category/category.py

```diff
@@ -0,0 +1,158 @@
+        self.model = self.load_model(model_path=model_path)
+
+    def load_model(self, model_path=None):
+        if model_path and os.path.isfile(model_path):
+            clf = joblib.load(model_path)
+        else:
+            default_model_path = 'category.pkl'
+            if os.path.isfile(default_model_path):
+                clf = joblib.load(default_model_path)
+            else:
+                url = 'https://s3-us-west-2.amazonaws.com/idmc-idetect/category_models/category.pkl'
+                r = requests.get(url, stream=True)
+                if not os.path.isfile(default_model_path):
+                    try:
+                        os.makedirs(os.path.dirname(default_model_path))
+                    except OSError as exc: # Guard against race condition
+                        if exc.errno != errno.EEXIST:
+                            raise
+                with open(default_model_path, 'wb') as f:
+                    for chunk in r.iter_content(chunk_size=1024):
+                        if chunk:  # filter out keep-alive new chunks
+                            f.write(chunk)
+                clf = joblib.load(default_model_path)
+        return clf
+
+    def predict(self, text):
+        try:
+            category = self.model.predict(pd.Series(text))[0]
+        except:
+            # if error occurs, classify as most likely category
+            category = 'disaster'
+
+        if category == 'disaster':
+            return Category.DISASTER
+        elif category == 'conflict':
+            return Category.CONFLICT
+        else:
+            return Category.OTHER
+
+class Tokenizer(TransformerMixin):
+    def __init__(self, stop_words=None):
+        self.stop_words = stop_words
+        self.tokenizer = WordPunctTokenizer()
+        
+    def prepare_tokens(self, text, stop_words):
+        tokens = self.tokenizer.tokenize(text)
+        tokens = [t for t in tokens if len(t) > 2]
+        tokens = [t for t in tokens if t not in stop_words]
+        tokens = [t.lower() for t in tokens]
+        tokens = [t for t in tokens if not t.isdigit()]
+        return tokens
+    
+    def fit(self, X, *args):
+        return self
+    
+    def transform(self, X, *args):
+        X = X.map(lambda x: self.prepare_tokens(x, self.stop_words))
+        return X
+
+class TfidfTransformer(TransformerMixin):
+    def __init__(self, no_below=5, no_above=0.5, tfidf_model=None, dictionary=None):
+        self.dictionary = dictionary
+        self.tfidf_model = tfidf_model
+        self.no_below = no_below
+        self.no_above = no_above
+    
+    def make_dictionary(self, texts):
+        if not self.dictionary:
+            self.dictionary = gensim.corpora.Dictionary(texts)
+            if self.no_below or self.no_above:
+                self.dictionary.filter_extremes(no_below=self.no_below, no_above=self.no_above)
+        return self
+    
+    def tfidf_transform(texts, dictionary=None, tfidf_model=None):
```

Comment:
It was getting called in the LSI Transformer, but was unnecessary. Have removed.